x/ibctransferhooks/keeper/keeper.go

```diff
@@ -0,0 +1,24 @@
+package keeper
+
+import "github.com/persistenceOne/persistence-sdk/x/ibctransferhooks/types"
+
+type Keeper struct {
+	hooks types.IBCTransferHooks
+}
+
+func NewKeeper() Keeper {
+	return Keeper{
+		hooks: nil,
+	}
+}
+
+// Set the validator hooks
+func (k *Keeper) SetHooks(transferHooks types.IBCTransferHooks) *Keeper {
+	if k.hooks != nil {
+		panic("cannot set hooks twice")
```

Comment:
since this is set only once, would it make sense to just pass this to NewKeeper itself.
```
func NewKeeper(transferHooks types.IBCTransferHooks) Keeper
```

Reply:
yes and no for this one, for this specific module, we can actually do it..
But this is not how usually it is done. 
And a module can be importing this keeper. which might also be implementing the hooks. 
that is why we have set hooks as different function